### PR TITLE
chore(lifecycle): add comments and CTEs to clickhouse sql

### PR DESCRIPTION
**Pull request by hazzadous**
It was really big and confusing, but hopefully this clarifies a little
what is going on. As a followup PR I'll be doing some work to make the
query faster :fingerscrossed: but I think worth at least getting this
in, assuming I haven't broken any tests!

## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

## How did you test this code?

*Please describe.*

### CODE REVIEW

File: ee/clickhouse/sql/trends/lifecycle.py [hunk 1]
```diff
@@ -1,59 +1,161 @@
...
+
+        WHERE has(person_activity.day, period_starts.day_start) = 0
+
+        ORDER BY 
+            person_id, 
+            period_starts.day_start ASC
+    ) 
+    
+    WHERE
+        -- exclude first period ofr dormant
```
Comment (EDsCODE) [inline ee/clickhouse/sql/trends/lifecycle.py:133]: type

File: ee/clickhouse/sql/trends/lifecycle.py [hunk 2]
```diff
@@ -1,59 +1,161 @@
...
+        day
+        
+    HAVING 
+        day <= toDateTime(%(date_to)s) 
+        AND day >= toDateTime(%(prev_date_from)s)
+
+), person_activity_as_array AS (
+    SELECT DISTINCT 
+        person_id, 
+        groupArray({trunc_func}(events.timestamp)) day
```
Comment (EDsCODE) [inline ee/clickhouse/sql/trends/lifecycle.py:24]: Can we change all "day" references to "period" or something generic so it accounts for week/months/hours

**Review by EDsCODE — COMMENTED** [review]
few suggestions

File: ee/clickhouse/sql/trends/lifecycle.py (hunk 2 above)
Comment (hazzadous) [reply ee/clickhouse/sql/trends/lifecycle.py]: have updated, what do you think?

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**